libraries/helpers.rb

```diff
@@ -30,8 +30,20 @@ def splunk_cmd
 
 def splunk_dir
   # Splunk Enterprise (Server) will install in /opt/splunk.
-  # Splunk Universal Forwarder (not Server) will install in /opt/splunkforwarder
-  node['splunk']['is_server'] ? '/opt/splunk' : '/opt/splunkforwarder'
+  # Splunk Universal Forwarder can be a used as a client or a forwarding
+  # (intermediary) server which installs to /opt/splunkforwarder
+  forwarderpath = '/opt/splunkforwarder'
+  enterprisepath = '/opt/splunk'
+  if node['splunk']['is_intermediate'] == true
+    path = forwarderpath
+    return path
+  elsif node['splunk']['is_server'] == true
+    path = enterprisepath
+    return path
+  else
+    path = forwarderpath
+    return path
+  end
```

Comment:
Is this conditional to catch `is_intermediate` is true when `is_server` also is set?

We can make use of Ruby's implicit returns from the conditional:

```ruby
  forwarderpath = '/opt/splunkforwarder'
  enterprisepath = '/opt/splunk'
  if node['splunk']['is_intermediate'] == true
    forwarderpath
  elsif node['splunk']['is_server'] == true
    enterprisepath
  else
    forwarderpath
  end
```

Implementation-wise this works:

```
chef:recipe > node.default['splunk']['is_intermediate'] = false
 => false
chef:recipe > node.default['splunk']['is_server'] = false
 => false
chef:recipe > splunk_dir
 => "/opt/splunkforwarder"
chef:recipe > node.default['splunk']['is_server'] = true
 => true
chef:recipe > splunk_dir
 => "/opt/splunk"
chef:recipe > node.default['splunk']['is_intermediate'] = true
 => true
chef:recipe > splunk_dir
 => "/opt/splunkforwarder"
```